graylog2-web-interface/src/components/nodes/JvmHeapUsage.jsx

```diff
@@ -1,102 +1,86 @@
-                detail = (<p>
-                    The JVM is using <span className="blob" style={{backgroundColor: "#9e1f63"}}></span>
-                    <strong>
-                        <span className="heap-used"> {numeral(this.state.usedMemory).format('0.0 b')}</span> of <span
-                        className="blob" style={{backgroundColor: "#f7941e"}}></span> <span
-                        className="heap-total"> {numeral(this.state.committedMemory).format('0.0 b')}</span></strong> heap space and will
-                    not attempt to use more than <span className="blob" style={{backgroundColor: "#f5f5f5"}}></span>
-                    <strong><span className="heap-max"> {numeral(this.state.maxMemory).format('0.0 b')}</span></strong>
-                </p>);
-            } else {
-                detail = (<p><i className="fa fa-spin fa-spinner"></i> Loading heap usage information...</p>);
-            }
-        }
+const JvmHeapUsage = React.createClass({
+  propTypes: {
+    nodeId: PropTypes.string.isRequired,
+  },
+  mixins: [Reflux.connect(MetricsStore)],
+  componentWillMount() {
+    this.metricNames = {
+      usedMemory: 'jvm.memory.heap.used',
+      committedMemory: 'jvm.memory.heap.committed',
+      maxMemory: 'jvm.memory.heap.max',
+    };
 
-        return (
-            <div className="graylog-node-heap" data-node-id={this.props.nodeId}>
-                <div className="progress">
-                    <div className="progress-bar heap-used-percent" style={{width: this.state.usedPercentage + "%"}}></div>
-                    <div className="progress-bar progress-bar-warning heap-total-percent" style={{width: (this.state.committedPercentage -this.state.usedPercentage) + "%"}}></div>
-                </div>
+    Object.keys(this.metricNames).forEach(metricShortName => MetricsActions.add(this.props.nodeId, this.metricNames[metricShortName]));
+  },
+  componentWillUnmount() {
+    Object.keys(this.metricNames).forEach(metricShortName => MetricsActions.remove(this.props.nodeId, this.metricNames[metricShortName]));
+  },
+  _extractMetricValues() {
+    const nodeId = this.props.nodeId;
+    const nodeMetrics = this.state.metrics[nodeId];
+    const metrics = MetricsExtractor.getValuesForNode(nodeMetrics, this.metricNames);
 
-                {detail}
-            </div>
+    metrics.usedPercentage = metrics.maxMemory === 0 ? 0 : (metrics.usedMemory / metrics.maxMemory) * 100;
+    metrics.committedPercentage = metrics.maxMemory === 0 ? 0 : (metrics.committedMemory / metrics.maxMemory) * 100;
+
+    return metrics;
+  },
+  render() {
+    let progressBar;
+    let detail;
+
+    if (this.state.metrics) {
+      const metrics = this._extractMetricValues();
+
+      if (Object.keys(metrics).length === 0) {
+        progressBar = <div className="progress"></div>;
+        detail = <p>Heap information unavailable.</p>;
+      } else {
+        progressBar = (
+          <div className="progress">
```

Comment:
I basically left the code as it was, will play a bit with `ProgressBar` and see if it makes it easier or better.